libagent/device/ledger.py

```diff
@@ -36,6 +36,19 @@ def _convert_public_key(ecdsa_curve_name, result):
 
 class LedgerNanoS(interface.Device):
     """Connection to Ledger Nano S device."""
+    ledger_app_name = b"SSH/PGP Agent"
+    ledger_app_version = ""
+    ledger_app_version_minor = 0
+
+    def get_app_name_and_version(self, dongle):
+        device_version_answer = dongle.exchange(binascii.unhexlify('B001000000'))
+        # check if we are running the ssh app on the Ledger device
+        app_name_length = device_version_answer[1]
```

Comment:
consider using https://docs.python.org/3/library/struct.html to unpack the `device_version_answer`